sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/HttpPipelineAdapter.java

```diff
@@ -57,4 +84,79 @@ public IHttpResponse send(HttpRequest httpRequest) throws Exception {
                 })))
             .block();
     }
+
+    private void logAccounIdentifiersIfConfigured(String body) {
+        if (identityClientOptions != null
+            && !identityClientOptions.getIdentityLogOptionsImpl().isLoggingAccountIdentifiersAllowed()) {
+            return;
+        }
+        try {
+            JsonParser responseParser = JSON_FACTORY.createParser(body);
+            String accessToken = getTargetFieldValueFromJsonParser(responseParser, ACCESS_TOKEN_JSON_KEY);
+            responseParser.close();
+            if (accessToken != null) {
+                String[] base64Metadata = accessToken.split("\\.");
+                if (base64Metadata.length > 1) {
+                    byte[] decoded = Base64.getDecoder().decode(base64Metadata[1]);
+                    String data = new String(decoded, StandardCharsets.UTF_8);
+                    JsonParser jsonParser = JSON_FACTORY.createParser(data);
+                    HashMap<String, String> jsonMap = parseJsonIntoMap(jsonParser);
+                    jsonParser.close();
+                    String appId = jsonMap.containsKey(APPLICATION_ID_JSON_KEY)
+                        ? jsonMap.get(APPLICATION_ID_JSON_KEY) : null;
+                    String objectId = jsonMap.containsKey(OBJECT_ID_JSON_KEY)
+                        ? jsonMap.get(OBJECT_ID_JSON_KEY) : null;
+                    String tenantId = jsonMap.containsKey(TENANT_ID_JSON_KEY)
+                        ? jsonMap.get(TENANT_ID_JSON_KEY) : null;
+                    String userPrincipalName = jsonMap.containsKey(USER_PRINCIPAL_NAME_JSON_KEY)
+                        ? jsonMap.get(USER_PRINCIPAL_NAME_JSON_KEY) : null;
+
+                    CLIENT_LOGGER.log(LogLevel.INFORMATIONAL, () -> MessageFormat
+                        .format(ACCOUNT_IDENTIFIER_LOG_MESSAGE,
+                            getAccountIdentifierMessage(APPLICATION_IDENTIFIER, appId),
```

Comment:
It looks like the fields are out of order with the fromat string. Should be `APPLICATION_IDENTIFIER`, `TENANT_ID`, `USER_PRINCIPAL_NAME`, `OBJECT_ID`?

Reply:
updated the order.